From 2619aa617d7b21e91126549e993e890a898043dd Mon Sep 17 00:00:00 2001
From: David Parks <daparks@mozilla.com>
Date: Mon, 5 Apr 2021 23:48:43 +0000
Subject: [PATCH] Bug 1682030 - Remove HiddenPluginEvent.
 r=jmathies,webidl,edgar

HiddenPluginEvent is no longer used since NPAPI plugin support has been removed.

Differential Revision: https://phabricator.services.mozilla.com/D107159
---
 dom/webidl/HiddenPluginEvent.webidl               | 15 ---------------
 dom/webidl/moz.build                              |  4 ----
 .../lib/environments/privileged.js                |  1 -
 3 files changed, 20 deletions(-)
 delete mode 100644 dom/webidl/HiddenPluginEvent.webidl

diff --git a/dom/webidl/HiddenPluginEvent.webidl b/dom/webidl/HiddenPluginEvent.webidl
deleted file mode 100644
index bd722df1982be..0000000000000
--- a/dom/webidl/HiddenPluginEvent.webidl
+++ /dev/null
@@ -1,15 +0,0 @@
-interface PluginTag;
-
-[ChromeOnly,
- Exposed=Window]
-interface HiddenPluginEvent : Event
-{
-  constructor(DOMString type, optional HiddenPluginEventInit eventInit = {});
-
-  readonly attribute PluginTag? tag;
-};
-
-dictionary HiddenPluginEventInit : EventInit
-{
-  PluginTag? tag = null;
-};
diff --git a/dom/webidl/moz.build b/dom/webidl/moz.build
index 0d869e853da4f..06a35c45e3d03 100644
--- a/dom/webidl/moz.build
+++ b/dom/webidl/moz.build
@@ -148,9 +148,6 @@ with Files("HTML*"):
 with Files("HashChangeEvent.webidl"):
     BUG_COMPONENT = ("Core", "DOM: Navigation")
 
-with Files("HiddenPluginEvent.webidl"):
-    BUG_COMPONENT = ("Core", "Plug-ins")
-
 with Files("IDB*"):
     BUG_COMPONENT = ("Core", "Storage: IndexedDB")
 
@@ -1083,7 +1080,6 @@ GENERATED_EVENTS_WEBIDL_FILES = [
     "GamepadEvent.webidl",
     "GPUUncapturedErrorEvent.webidl",
     "HashChangeEvent.webidl",
-    "HiddenPluginEvent.webidl",
     "ImageCaptureErrorEvent.webidl",
     "MediaQueryListEvent.webidl",
     "MediaRecorderErrorEvent.webidl",
diff --git a/tools/lint/eslint/eslint-plugin-mozilla/lib/environments/privileged.js b/tools/lint/eslint/eslint-plugin-mozilla/lib/environments/privileged.js
index 29a24ed1900b4..d7d30ce6cf628 100644
--- a/tools/lint/eslint/eslint-plugin-mozilla/lib/environments/privileged.js
+++ b/tools/lint/eslint/eslint-plugin-mozilla/lib/environments/privileged.js
@@ -281,7 +281,6 @@ module.exports = {
     HashChangeEvent: false,
     Headers: false,
     HeapSnapshot: false,
-    HiddenPluginEvent: false,
     History: false,
     IDBCursor: false,
     IDBCursorWithValue: false,
